### PR TITLE
test(block): integrated and added test cases for component

### PR DESCRIPTION
**Pull request by timrombergjakobsson**

### CODE REVIEW

File: src/components/Block.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+const Block = () => {
+    return (
+        <div>
+            <div className="tds-headline-02 tds-u-pb1">Block</div>
```
Comment (timrombergjakobsson) [inline src/components/Block.tsx:4]: Should I add the different variants of the Block?

Comment (mJarsater) [reply src/components/Block.tsx:4]: Sure let's!

File: src/components/Block.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+          radio-id="variant-option-2"
+          checked={modeVariant === 'secondary'}
+					onTdsChange={handleVariantChange}
+        >
+          <div slot="label">Secondary variant</div>
+        </tds-radio-button>
+      </div>
+
+      <tds-block mode-variant={modeVariant}>
+			<div className="tds-headline-02 tds-u-pb1">{capitalizeFirst(modeVariant)} Block</div>
```
Comment (timrombergjakobsson) [inline src/components/Block.tsx:64]: I miss Rubys `capitalize()`

**Review by mJarsater — APPROVED** [review]
LGTM

File: src/components/Block.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    };
+
+    tdsRadioButtons.forEach((tdsRadioButton) => {
+      tdsRadioButton.addEventListener('tdsChange', handleChange);
+
+      return () => {
+        tdsRadioButton.removeEventListener('tdsChange', handleChange);
+      };
+    });
+  }, []);
```
Comment (mJarsater) [inline src/components/Block.tsx:36]: I think these should be set on the element inside the runder function instead, in the way we show in our docs.

File: src/components/Block.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+  return (
+    <div>
+      <div className="tds-headline-02 tds-u-pb1">Block</div>
+      <div>
+        <tds-radio-button
```
Comment (mJarsater) [inline src/components/Block.tsx:42]:
```suggestion
        <tds-radio-button
        ref={(element) => element.addEventListener('tdsChange' , () => {
          setModeVariant(element.modeVariant);
        })}
```

File: src/components/Block.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect } from 'react';
+
+interface TdsRadioButtonCustomEvent<T> extends Event {
+  target: EventTarget & T;
+}
+
+const Block: React.FC = () => {
```
Comment (mJarsater) [inline src/components/Block.tsx:7]:
```suggestion
const Block = () => {
```

**Review by mJarsater — CHANGES_REQUESTED** [review]
Some minor comments

**Review by mJarsater — APPROVED** [review]
Lovely dubly!

**Pull request merged**